src/uct/self/self_iface.h

```diff
@@ -10,13 +10,21 @@
 
 typedef uint64_t uct_self_iface_addr_t;
 
+typedef struct uct_self_msg_desc {
+    uct_am_recv_desc_t am_recv_desc;    /* Used to reserve descriptor */
+    uct_am_recv_desc_t am_recv_data[0]; /* Payload */
+} uct_self_msg_desc_t;
```

Comment:
no need to define special type, can just use uct_am_recv_desc_t instead of uct_self_msg_desc_t